frontend/src/components/Order.js

```diff
@@ -55,34 +105,68 @@ function Order({ order, orderNr }) {
-                }
-                // eslint-disable-next-line no-param-reassign
-                item.qty = 1;
-                return acc.concat(item);
-              }, [])
-              .map((orderedItem) =>
-                orderedItem.type === "product" ? (
+            {mutableItems.map((orderedItem) =>
+              orderedItem.type === "product" ? (
+                <View
+                  flexDirection="row"
+                  justifyContent="space-between"
+                  key={orderedItem._id}
+                >
+                  <Text color="text.light" mt="2" fontSize="sm">
+                    {orderedItem.cartQty}x{" "}
+                  </Text>
+                  <Text color="text.light" mt="2" fontSize="sm">
+                    {orderedItem.name}
+                  </Text>
+                  <Text
+                    color="text.light"
+                    mt="2"
+                    fontSize="sm"
+                    flexGrow={1}
+                    textAlign="end"
+                    style={{ whiteSpace: "nowrap" }}
+                  >
+                    {orderedItem.price * orderedItem.cartQty} RON
+                  </Text>
+                </View>
+              ) : (
+                <View
+                  flexDirection="row"
+                  justifyContent="space-between"
+                  key={`${orderedItem._id}_${orderedItem.recipe[0].name}`}
+                >
+                  <Text color="text.light" mt="2" fontSize="sm">
+                    {orderedItem.cartQty}x{" "}
+                  </Text>
                   <View
-                    flexDirection="row"
+                    flexDirection="column"
                     justifyContent="space-between"
-                    key={orderedItem._id}
+                    flexGrow={1}
                   >
                     <Text color="text.light" mt="2" fontSize="sm">
-                      {orderedItem.qty}x {orderedItem.name}
+                      {orderedItem.name}
                     </Text>
 
-                    <Text color="text.light" mt="2" fontSize="sm">
-                      {orderedItem.price * orderedItem.qty} RON
+                    <Text
+                      color="text.light"
+                      mt="2"
+                      fontSize="xs"
+                      marginTop="0.2rem"
+                    >
+                      -{orderedItem.recipe[0].name}
                     </Text>
                   </View>
-                ) : null
-              )}
+                  <Text
+                    color="text.light"
+                    mt="2"
+                    fontSize="sm"
+                    style={{ whiteSpace: "nowrap" }}
+                  >
+                    {orderedItem.recipe[0].price * orderedItem.cartQty} RON
+                  </Text>
+                </View>
+              )
+            )}
 
             <Divider bg="transparent" thickness="10" />
             <Divider bg="black" thickness="1" />
```

Comment:
The code patch disables several ESLint rules, which might make it harder to catch common coding mistakes or best practices violations. 

Regarding the `composeItemsFromVariations` function, some things to note are:
- It has a good descriptive name that explains what it does.
- It makes changes to the input `order` object, which could be surprising for other parts of the codebase. A better approach would be to clone the input object and return a new, updated one.
- The function has several nested loops, which can decrease performance if the input data is large. Consider optimizing or refactoring if this becomes an issue.
- The code that updates `existingItem.cartQty` looks duplicated and could be extracted into its own function.

In the `Order` component:
- The code calls the `composeItemsFromVariations` function instead of making changes to a duplicate of the input `order`.
- There's duplicated code inside the `.map()` method calls for product and recipe items that could be refactored.
- This component also lacks proper error handling, which could lead to unexpected behavior if passed invalid data.

Overall, suggestions for improvement include:
- Instead of disabling ESLint rules, fix any issues they're warning about.
- Use a functional style instead of modifying input parameters in-place, as it is easier to reason about and avoids unexpected side-effects.
- Move duplicated code into their own functions to improve maintainability.
- Consider adding more structured error handling.